Sources/MessagingPush/PushHandling/NotificationCenterFrameworkAdapter.swift

```diff
@@ -61,6 +86,47 @@ class NotificationCenterFrameworkAdapterImpl: NSObject, UNUserNotificationCenter
             completionHandler([])
         }
     }
+
+    // Swizzle method convenient when original and swizzled methods both belong to same class.
+    func swizzle(forClass: AnyClass, original: Selector, new: Selector) {
+        guard let originalMethod = class_getInstanceMethod(forClass, original) else { return }
+        guard let swizzledMethod = class_getInstanceMethod(forClass, new) else { return }
+
+        let didAddMethod = class_addMethod(forClass, original, method_getImplementation(swizzledMethod), method_getTypeEncoding(swizzledMethod))
+
+        if didAddMethod {
+            class_replaceMethod(forClass, new, method_getImplementation(originalMethod), method_getTypeEncoding(originalMethod))
+        } else {
+            method_exchangeImplementations(originalMethod, swizzledMethod)
+        }
+    }
+}
+
+// This is a bit confusing and makes the code a little more complex. However, it's the most reliable way found to get UNUserNotificationCenter.delegate swizzling to work by using an extension.
+@available(iOSApplicationExtension, unavailable)
+extension UNUserNotificationCenter {
+    // Swizzled method that gets called when `UNUserNotificationCenter.current().delegate` setter called.
+    @objc dynamic func cio_swizzled_setDelegate(delegate: UNUserNotificationCenterDelegate?) {
+        guard let diGraph = SdkInitializedUtilImpl().postInitializedData?.diGraph else {
+            // SDK not initialized yet. We hope that this function never gets called because auto push click handling will not work if this function called.
+
+            // Forward request to the original implementation implementation that we swizzled. So that the app finishes setting UNUserNotificationCenter.delegate.
```

Comment:
```suggestion
            // Forward request to the original implementation that we swizzled. So that the app finishes setting UNUserNotificationCenter.delegate.
```